### PR TITLE
Add build on travis-ci and deployment to Github

### PR DESCRIPTION
**Pull request by JohanLorenzo**

### CODE REVIEW

**Comment by JohanLorenzo** [comment]
I thought you could have builds without having any travis-ci in the upstream/master. I don't see any job running at https://travis-ci.org/mozilla/testdroid-marionette-plugin/pull_requests . So this'll be a blind review :( 
@davehunt r?

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+jdk:
+- openjdk7
+
+script:
+- mvn clean package
+
+deploy:
+  provider: releases
+  api_key:
+    secure: AuN0zYIV1UQqEAc9kdxHaeP5gTFD0WFGyf777/JEp2VhoCbBuSldlfXy7r4ELK8kqr1OostlsVGIFq0bKUB9bAYTssfom3/tz2UdVtZl5BR/nJGW5eDRUcJNBG+CTVlrnsGP08eXwNjGUvqGMwVUZbXdJLYJGCU6Kh06k76x7kE=
```
Comment (JohanLorenzo) [inline .travis.yml]: I generated this token from the Travis-Cl CLI. The command was decribed there http://docs.travis-ci.com/user/deployment/releases/#Authenticating-with-an-Oauth-token

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  file: target/testdroid-marionette.hpi
+  on:
+    repo: mozilla/testdroid-marionette-plugin
```
Comment (JohanLorenzo) [inline .travis.yml]: Don't deploy anything on a fork.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    tags: true
```
Comment (JohanLorenzo) [inline .travis.yml]: Don't deploy regular commits. Just the commits that are tagged

**Comment by davehunt** [comment]
Closed/reopened to initiate a Travis CI build.

**Comment by davehunt** [comment]
Looks good, although the deploy stuff is new to me. Could you add the Maven dependencies to cache so we don't download them each time? See http://docs.travis-ci.com/user/caching/ and I think you want $HOME/.m2

**Comment by JohanLorenzo** [comment]
Good idea! I added that. Let's see how Travis behaves.

**Comment by JohanLorenzo** [comment]
First build that created the cache: https://travis-ci.org/mozilla/testdroid-marionette-plugin/builds/74421746#L3485
Second build which uses the cache: https://travis-ci.org/mozilla/testdroid-marionette-plugin/builds/74423096#L87 and updates it: https://travis-ci.org/mozilla/testdroid-marionette-plugin/builds/74423096#L277

It looks good to me. Let's merge it try the deployment of the first version.

**Comment by JohanLorenzo** [comment]
*The configuration has been verified.*

After the merge, the build on the master branch didn't publish the package (like expected): https://travis-ci.org/mozilla/testdroid-marionette-plugin/builds/74424162. 

After pushing the tag v0.0.1, Travis did deploy to GitHub: https://travis-ci.org/mozilla/testdroid-marionette-plugin/builds/74426758#L300 https://github.com/mozilla/testdroid-marionette-plugin/releases

**Pull request merged**